agent/dockerclient/dockerauth/asm_error.go

```diff
@@ -0,0 +1,34 @@
+// Copyright 2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package dockerauth
+
+// ASMAuthDataError is used to encapsulate the error returned...
+type ASMAuthDataError struct {
```

Comment:
Does this error needs to implement the retrier to make sure this error isn't retried?

Reply:
it should, it implements the Retry() method and returns false.